pkg/remote/aws/iam_role_supplier.go

```diff
@@ -40,7 +46,7 @@ func awsIamRoleShouldBeIgnored(roleName string) bool {
 }
 
 func (s IamRoleSupplier) Resources() ([]resource.Resource, error) {
-	roles, err := listIamRoles(s.client)
+	roles, err := listIamRoles(s.client, resourceaws.AwsIamRoleResourceType)
```

Comment:
Maybe we could add a method to resource supplier interface `SupportedTypes() []string` ? It will be usefull in the future to avoid run filtered suppliers. Could you open a refactoring issues with this details if you agree ?

Reply:
yeah I was thinking about that but the pr was already too large :sweat_smile:

Reply:
https://github.com/cloudskiff/driftctl/issues/160

Reply:
Thanks !